tests/integration/golang/mlflow/artifact/get_artifact_local_test.go

```diff
@@ -0,0 +1,145 @@
+//go:build integration
+
+package artifact
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/hetiansu5/urlquery"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow"
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/api/request"
+	"github.com/G-Research/fasttrackml/pkg/api/mlflow/dao/models"
+	"github.com/G-Research/fasttrackml/tests/integration/golang/fixtures"
+	"github.com/G-Research/fasttrackml/tests/integration/golang/helpers"
+)
+
+type GetArtifactLocalTestSuite struct {
+	suite.Suite
+	runFixtures        *fixtures.RunFixtures
+	serviceClient      *helpers.HttpClient
+	experimentFixtures *fixtures.ExperimentFixtures
+}
+
+func TestGetArtifactLocalTestSuite(t *testing.T) {
+	suite.Run(t, new(GetArtifactLocalTestSuite))
+}
+
+func (s *GetArtifactLocalTestSuite) SetupTest() {
+	s.serviceClient = helpers.NewMlflowApiClient(helpers.GetServiceUri())
+
+	experimentFixtures, err := fixtures.NewExperimentFixtures(helpers.GetDatabaseUri())
+	assert.Nil(s.T(), err)
+	s.experimentFixtures = experimentFixtures
+
+	runFixtures, err := fixtures.NewRunFixtures(helpers.GetDatabaseUri())
+	assert.Nil(s.T(), err)
+	s.runFixtures = runFixtures
+}
+
+func (s *GetArtifactLocalTestSuite) Test_Ok() {
+	defer func() {
+		assert.Nil(s.T(), s.experimentFixtures.UnloadFixtures())
+	}()
+
+	testData := []struct {
+		name   string
+		prefix string
+	}{
+		{
+			name:   "TestWithFilePrefix",
+			prefix: "file://",
+		},
+		{
+			name:   "TestWithoutPrefix",
+			prefix: "",
+		},
+	}
+
+	for _, tt := range testData {
+		s.T().Run(tt.name, func(t *testing.T) {
+			// 1. create test experiment.
+			experimentArtifactDir := t.TempDir()
+			experiment, err := s.experimentFixtures.CreateExperiment(context.Background(), &models.Experiment{
+				Name: fmt.Sprintf("Test Experiment In Path %s", experimentArtifactDir),
+				Tags: []models.ExperimentTag{
+					{
+						Key:   "key1",
+						Value: "value1",
+					},
+				},
+				CreationTime: sql.NullInt64{
+					Int64: time.Now().UTC().UnixMilli(),
+					Valid: true,
+				},
+				LastUpdateTime: sql.NullInt64{
+					Int64: time.Now().UTC().UnixMilli(),
+					Valid: true,
+				},
+				LifecycleStage:   models.LifecycleStageActive,
+				ArtifactLocation: fmt.Sprintf("%s%s", tt.prefix, experimentArtifactDir),
+			})
+			assert.Nil(s.T(), err)
+
+			// 2. create test run.
+			runID := strings.ReplaceAll(uuid.New().String(), "-", "")
+			runArtifactDir := filepath.Join(experimentArtifactDir, runID, "artifacts")
+			run, err := s.runFixtures.CreateRun(context.Background(), &models.Run{
+				ID:             runID,
+				Status:         models.StatusRunning,
```

Comment:
do we really need to do all of that in cycle?

Reply:
not clear what `cycle` is here? Do you mean unnecessary attributes for the test?